src/include/duckdb/main/client_context.hpp

```diff
@@ -89,6 +89,8 @@ class ClientContext : public enable_shared_from_this<ClientContext> {
 	//! Data for the currently running transaction
 	TransactionContext transaction;
 
+	duckdb_progress_callback_t progress_callback = nullptr;
```

Comment:
It's a little weird to me to marry the C API to the C++ API like this, I think we want to create a c version and a c++ version of this. In `duckdb_set_progress_callback`  you would wrap the c version in the c++ version instead.